### PR TITLE
Add communi IRC client

### PR DESCRIPTION
**Pull request by hrdinka**
This adds libcommuni as well as the communi desktop IRC client. The configure script might look a bit curious but that is how it is actually supposed to be done:

https://github.com/communi/communi-desktop/wiki/How-to-build-communi-in-its-own-directory

I have decided to use the latest git commits as versions instead of stable releases. The official guide does only show how to install communi from its master branch. I guess that is because there isn't much release management going on—there have only been 4 release since 2008.

### CODE REVIEW

File: pkgs/applications/networking/irc/communi/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{ fetchgit, libcommuni, qt5, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "communi-${version}";
+  version = "ad1b9a30ed";
```
Comment (jagajaga) [inline pkgs/applications/networking/irc/communi/default.nix]: Please change version to date. See https://nixos.org/nixpkgs/manual/#sec-package-naming

File: pkgs/development/libraries/libcommuni/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ fetchgit, qt5, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "libcommuni-${version}";
+  version = "779b0c7744";
```
Comment (jagajaga) [inline pkgs/development/libraries/libcommuni/default.nix]: Please change version to date. See https://nixos.org/nixpkgs/manual/#sec-package-naming

Comment (hrdinka) [inline pkgs/development/libraries/libcommuni/default.nix]: Thanks for the hint. Done.

File: pkgs/applications/networking/irc/communi/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{ fetchgit, libcommuni, qt5, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "communi-${version}";
+  version = "git-2016-01-03";
```
Comment (jagajaga) [inline pkgs/applications/networking/irc/communi/default.nix]: You can omit `git` prefix now.. That will be a better way.

File: pkgs/development/libraries/libcommuni/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{ fetchgit, qt5, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  name = "libcommuni-${version}";
+  version = "git-2016-01-02";
```
Comment (jagajaga) [inline pkgs/development/libraries/libcommuni/default.nix]: Same here.

File: pkgs/applications/networking/irc/communi/default.nix (hunk 3 above)
Comment (hrdinka) [inline pkgs/applications/networking/irc/communi/default.nix]: Done. Better just for this package or generally?

**Pull request merged**